domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsEquivalentToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,318 @@
+    val inpus = mapOf("f" to FRACTION_VALUE_242_OVER_1)
+    val answer = FRACTION_VALUE_33_OVER_2
+
+    val matches =
+      inputIsEquivalentToRuleClassifier.matches(answer = FRACTION_VALUE_33_OVER_2, inputs = inpus)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquality_mixedNumbr6And1Over2Answer_withMixedNumber6And1Over2Input_bothValuesMatch() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_6_1_OVER_2)
+
+    val matches =
+      inputIsEquivalentToRuleClassifier.matches(
+        answer = MIXED_NUMBER_VALUE_6_1_OVER_2,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquality_mixedNumber55And1Over4Answer_withMixedNum6And1Over2Input_bothValuesDoNotMatch() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_6_1_OVER_2)
+
+    val matches =
+      inputIsEquivalentToRuleClassifier.matches(
+        answer = MIXED_NUMBER_VALUE_55_1_OVER_4,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquality_wholeNumberAnswer_withMixedNumberInput_bothValuesDoNotMatch() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_6_1_OVER_2)
+
+    val matches =
+      inputIsEquivalentToRuleClassifier.matches(answer = WHOLE_NUMBER_VALUE_254, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquality_negativeMixedNumberAnswer_withPositveMixedNumberInput_bothValuesDoNotMatche() {
+    val inputs = mapOf("f" to NON_NEGATIVE_VALUE_679_2_OVER_3)
+
+    val matches =
+      inputIsEquivalentToRuleClassifier.matches(
+        answer = MIXED_NUMBER_VALUE_55_1_OVER_4,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_mixedNumberAnswer_withFractionInput_bothValuesDoNotMatch() {
+    val inputs = mapOf("f" to FRACTION_VALUE_1_OVER_5)
+
+    val matches =
+      inputIsEquivalentToRuleClassifier.matches(
+        answer = MIXED_NUMBER_VALUE_6_1_OVER_2,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_missingInputF_throwsException() {
+    val inputs = mapOf("y" to FRACTION_VALUE_2_OVER_8)
+
+    val exception = assertThrows(IllegalStateException::class) {
+      inputIsEquivalentToRuleClassifier.matches(answer = FRACTION_VALUE_2_OVER_8, inputs = inputs)
+    }
+
+    assertThat(exception)
+      .hasMessageThat()
+      .contains("Expected classifier inputs to contain parameter with name 'f' but had: [y]")
+  }
+
+  @Test
+  fun testEquality_nonNegativeInput_inputWithIncorrectType_throwsException() {
+    val inputs = mapOf("f" to NON_NEGATIVE_VALUE_0)
+
+    val exception = assertThrows(IllegalStateException::class) {
+      inputIsEquivalentToRuleClassifier.matches(answer = FRACTION_VALUE_2_OVER_8, inputs = inputs)
+    }
+
+    assertThat(exception)
+      .hasMessageThat()
+      .contains(
+        "Expected input value to be of type FRACTION not NON_NEGATIVE_INT"
+      )
+  }
+
+  private fun createFraction(
```

Comment:
This function is already available in `InteractionObjectTestBuilder`

Reply:
Yes it is in InteractionObjectTestBuilder

Reply:
Done